aten/src/ATen/core/function_schema.h

```diff
@@ -81,7 +81,12 @@ struct Argument {
         "' but instead found type '",
         actual_type,
         "'.\n",
-        inferred_type_hint);
+        inferred_type_hint,
```

Comment:
This error message is indeed more actionable, but it doesn't provide additional information or addresses user confusion (difference between 2 and 2.0). Maybe be explicit in suggesting user to check for integer vs float literals?

Reply:
@gmagogsfm : Can you please take a look at this fix?

Reply:
Optional:
There isn't a case where both inferred_type_hint and err_msg_for_int_float_literals are needed, you can replace both of them with a single variable, like "type_error_hint"